CS3230Project/Model/Users/Patients/PatientManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using CS3230Project.DAL.Patients;
+using CS3230Project.ErrorMessages;
+
+namespace CS3230Project.Model.Users.Patients
+{
+    /// <summary>
+    ///   Manages access and manipulation of the patients
+    /// </summary>
+    public static class PatientManager
+    {
+
+        /// <summary>
+        /// Gets all the patients with the provided name
+        /// Precondition:
+        ///     (firstName != null AND firstName.Trim().Length MORE THAN 0) ||
+        ///         (lastName != null AND lastName.Trim().Length MORE THAN 0)
+        /// </summary>
+        /// <param name="firstName">The first name.</param>
+        /// <param name="lastName">The last name.</param>
+        /// <returns>
+        ///     Returns a list of the patients with the provided name.
+        ///     Returns null if no name provided.
+        ///     Returns an empty list if there are no patients with the provided name.
+        /// </returns>
+        public static List<Patient> GetPatientsByName(string firstName, string lastName)
+        {
+            if ((firstName == null || firstName.Trim().Length == 0) &&
```

Comment:
I would separate out the precondition checks. From a users point of view they don't know what null really is. So when they see a message that something is null or empty it can be a bit confusion. I also like my error messages to be as clear as possible and point exactly to what is wrong. Does not have to hold up the approval but a separate ticket would need to be made if this is something that is easier for us to work on later.

Reply:
So I would throw separate argument exceptions for null and for empty. With separate error messages.